integration-tests/validatePIIRequest.spec.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  getLatestLogStreamName,
+  getMatchingLogEvents,
+  extractRequestID
+} from './utils/helpers'
+import { FilteredLogEvent } from '@aws-sdk/client-cloudwatch-logs'
+
+import { createZendeskRequest } from './utils/raiseZendeskRequest'
+import { approveZendeskRequest } from './utils/approveZendeskRequest'
+
+describe('Submit a PII request with approved ticket data', () => {
+  jest.setTimeout(45000)
+
+  it('Should log a success in cloud watch if Zendesk request is valid', async () => {
+    const ticketID = await createZendeskRequest(true)
+
+    await approveZendeskRequest(ticketID)
+
+    // Cloudwatch - Fetch latest log stream containing the ticket details
+    let latestLogStreamName = await getLatestLogStreamName()
+    console.log(`LATEST LOG STREAM NAME: ${latestLogStreamName}`)
+    let logStreamRequestID
+    let eventMatched = false
+
+    while (!eventMatched) {
```

Comment:
Is it worth breaking out waiting on a matching event into a separate function which returns a promise we can await on?

Reply:
Yes, I would agree, especially after I notice that you're doing exactly the same thing twice in both tests, the code looks to be duplicated, so should be put out into another function for that reason too.

Reply:
Done